features/dd-sdk-android-trace/src/main/kotlin/com/datadog/android/trace/internal/data/OtelTraceWriter.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.trace.internal.data
+
+import androidx.annotation.WorkerThread
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.context.DatadogContext
+import com.datadog.android.api.feature.Feature
+import com.datadog.android.api.feature.FeatureSdkCore
+import com.datadog.android.api.storage.EventBatchWriter
+import com.datadog.android.api.storage.RawBatchEvent
+import com.datadog.android.event.EventMapper
+import com.datadog.android.event.NoOpEventMapper
+import com.datadog.android.trace.internal.domain.event.ContextAwareMapper
+import com.datadog.android.trace.internal.storage.ContextAwareSerializer
+import com.datadog.android.trace.model.SpanEvent
+import com.datadog.trace.common.writer.Writer
+import com.datadog.trace.core.DDSpan
+import java.util.Locale
+
+internal class OtelTraceWriter(
+    private val sdkCore: FeatureSdkCore,
+    internal val legacyMapper: ContextAwareMapper<DDSpan, SpanEvent>,
+    internal val eventMapper: EventMapper<SpanEvent> = NoOpEventMapper(),
+    private val serializer: ContextAwareSerializer<SpanEvent>,
+    private val internalLogger: InternalLogger
+) : Writer {
+
+    // region Writer
+    override fun start() {
+        // NO - OP
+    }
+
+    override fun write(trace: MutableList<DDSpan>?) {
+        if (trace == null) return
```

Comment:
can it be null or just Kotlin inferred argument type as nullable by default?

Reply:
I need to check that

Reply:
well it is not marked as @NonNull in the interface so I wanted just to make sure. Probably it will never be send as null but don't want to touch the Java code for now.